gtsam/slam/slam.i

```diff
@@ -322,6 +322,17 @@ template <T = {gtsam::Point2, gtsam::Rot2, gtsam::Pose2, gtsam::Point3,
 virtual class KarcherMeanFactor : gtsam::NonlinearFactor {
   KarcherMeanFactor(const gtsam::KeyVector& keys);
 };
+  
+class Rot3Vector {
+  Rot3Vector();
+  size_t size() const;
+
+  // structure specific methods
+  gtsam::Rot3 at(size_t i) const;
+  void push_back(const gtsam::Rot3& R);
+};
+typedef FindKarcherMean<gtsam::Rot3> FindKarcherMeanRot3;
+gtsam::Rot3 FindKarcherMeanRot3(const Rot3Vector& rotations);
```

Comment:
@dellaert @ProfFan the CI states: `Cannot find class FindKarcherMean in module!`. Strange. Should we be including `#include <gtsam/slam/KarcherMeanFactor.h>` instead of `#include <gtsam/slam/KarcherMeanFactor-inl.h>` in this `.i` file?

```
Traceback (most recent call last):
  File "/home/runner/work/gtsam/gtsam/wrap/scripts/pybind_wrap.py", line 94, in <module>
    main()
  File "/home/runner/work/gtsam/gtsam/wrap/scripts/pybind_wrap.py", line 85, in main
    wrapper.wrap_submodule(args.src)
  File "/home/runner/work/gtsam/gtsam/wrap/gtwrap/pybind_wrapper.py", line 696, in wrap_submodule
    cc_content = self.wrap_file(content, module_name=module_name)
  File "/home/runner/work/gtsam/gtsam/wrap/gtwrap/pybind_wrapper.py", line 626, in wrap_file
    module = instantiator.instantiate_namespace(module)
  File "/home/runner/work/gtsam/gtsam/wrap/gtwrap/template_instantiator/namespace.py", line 80, in instantiate_namespace
    element = instantiate_namespace(element)
  File "/home/runner/work/gtsam/gtsam/wrap/gtwrap/template_instantiator/namespace.py", line 59, in instantiate_namespace
    typedef_inst.typename)
  File "/home/runner/work/gtsam/gtsam/wrap/gtwrap/interface_parser/namespace.py", line 109, in find_class_or_function
    typename.name))
ValueError: Cannot find class FindKarcherMean in module!
make[2]: *** [python/slam.cpp] Error 1
python/CMakeFiles/pybind_wrap_gtsam.dir/build.make:164: recipe for target 'python/slam.cpp' failed
CMakeFiles/Makefile2:6769: recipe for target 'python/CMakeFiles/pybind_wrap_gtsam.dir/all' failed
make[1]: *** [python/CMakeFiles/pybind_wrap_gtsam.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Reply:
But you *can* use `std::`. Who is telling you you can't ?

Reply:
Matlab wrapper needs correct signature, unless @varunagrawal implemented something that handles this?

Reply:
If I'm not mistaken, std should be useable. When in doubt, check out or write a test for it in the wrap project. :)

(I spent a whole month writing those tests folks, please make them worth my time 😅)

Reply:
> Matlab wrapper needs correct signature, unless @varunagrawal implemented something that handles this?

I don't know what that means. Example?